logic/observer.js

```diff
@@ -38,10 +39,21 @@ class Observer {
     }
 
     subscribe(subscriptionParams, user) {
-        //TODO: prevent duplicate subscriptions by checking subscription hash (fields "account", "asset_type" etc.)
-        //https://www.npmjs.com/package/farmhash
         return this.loadSubscriptions()
             .then(() => {
+                // Create hash in the subscription to avoid duplication
+    
+                let hashData = `${subscriptionParams.account} ${subscriptionParams.asset_type} ${subscriptionParams.asset_code} ${subscriptionParams.asset_issuer}`
```

Comment:
Please check all meaningful subscription params here: https://github.com/stellar-expert/operations-notifier/blob/93aa567c72c4c8a17cc2e14308ee80f9ef467326/logic/storage.js#L138

We can use `account`, `memo`, `operation_types`, `asset_code`, `expires`, and `reaction_url`. Maybe it makes sense to do something like `JSON.stringify(subscriptionParams)` to transparently handle all future parameters changes. What do you think?

Reply:
You are welcome!

-  Adding the relevant parameters for the hash --> will do

- having the hash as the generic_id --> I am in favor of it. However, I think that we need to be able to deal with older versions that will GET and DELETE based on the old id's. How do we deal with that? Finally, the DELETE endpoint does not DELETE the subscription from mongo, I could not find the code for it. Is this correct?

Reply:
I did a bit more of research and found that Mongo uses 24 character hex string for its _id. The hash generated is a 32 character hex string. No idea how to use the has as the generic id

Reply:
And about the delete endpoint I saw that the deleted subscriptions have a status of 1 👍